src/view.c

```diff
@@ -268,6 +261,38 @@ view_compute_centered_position(struct view *view, int w, int h, int *x, int *y)
 	return true;
 }
 
+static void
+set_fallback_geometry(struct view *view)
+{
+	view->natural_geometry.width = LAB_FALLBACK_WIDTH;
+	view->natural_geometry.height = LAB_FALLBACK_HEIGHT;
+	view_compute_centered_position(view,
+		view->natural_geometry.width,
+		view->natural_geometry.height,
+		&view->natural_geometry.x,
+		&view->natural_geometry.y);
+}
+#undef LAB_FALLBACK_WIDTH
+#undef LAB_FALLBACK_HEIGHT
+
+static void
+view_store_natural_geometry(struct view *view)
+{
+	/**
+	 * If an application was started maximized or fullscreened, its
+	 * natural_geometry width/height may still be zero in which case we set
+	 * some fallback values. This is the case with foot and Qt applications.
+	 */
+	if (!view->w && !view->h) {
```

Comment:
Do we want to keep the `&&` here or instead use `||` ?

Reply:
Yes, `||` makes more sense.
I don't suspect it'll make a difference in reality, but `||` is good.

Reply:
changed to `||` and rebased on latest master